src/scripts/modules/gradebook.js

```diff
@@ -295,59 +352,61 @@
 
-                if ( !IGNORED_GRADES.includes( submission.grade ) ) {
-                    submission.assignment = assignment;
-                    let existingSubmission = students[ studentId ].submissions[ assignment.id ];
-                    if ( !( existingSubmission && existingSubmission.submitted_at > submission.submitted_at ) ) {
-                      students[ studentId ].submissions[ assignment.id ] = submission;
-                    }
+                  if ( !IGNORED_GRADES.includes( submission.grade ) ) {
+                      submission.assignment = assignment;
+                      let existingSubmission = students[ studentId ].submissions[ assignment.id ];
+                      if ( !( existingSubmission && existingSubmission.submitted_at > submission.submitted_at ) ) {
+                        students[ studentId ].submissions[ assignment.id ] = submission;
+                      }
+                  }
+                } else {
+                  console.log("Submission from other path", submission, students[ studentId ])
                 }
-
             } );
 
         }
-
-        Promise.all( [ sessionData.students ] ).then( ( [ s ] ) => {
-            let students = {};
-            let assignments = [];
-
-            Promise.all( s.map( s => new Promise( ( res ) => {
-                $.get( userURI(s.id) ).then( html => res( scrapeStudentPage( students, assignments, userURI(s.id), html ) ) );
-            } ) ) ).then( () => {
-
-                assignments.sort( function( a, b ) {
-                    return new Date( b.first_submission_at ) - new Date( a.first_submission_at );
-                } );
-
-                students = calculateGrades( students, assignments );
-                const gradebook = {
-                    id: sessionData.path.id,
-                    title: sessionData.path.title,
-                    students: students,
-                    assignments: assignments,
-                    scraped_at:  moment()
-                };
-                let gradebooks = loadCachedGradebooks();
-
-                gradebooks[sessionData.path.id] = gradebook;
-
-                localStorage.setItem( 'cachedGradeBookData', JSON.stringify( gradebooks ) );
-                callback( gradebook );
-            } );
-        } );
+        Promise.all( [getPathContent(sessionData.path)] ).then( ([pathWithContent]) => {
+          sessionData.path = pathWithContent;
+
+          Promise.all( [sessionData.students] ).then( ( [ s ] ) => {
+              let students = {};
+              let assignments = sessionData.path.content.assignments;
+
+              Promise.all( s.map( s => new Promise( ( res ) => {
+                  $.get( userURI(s.id) ).then( html => res( scrapeStudentPage( students, assignments, userURI(s.id), html ) ) );
+              } ) ) ).then( () => {
+
+                  // Reject any assingments that have nothing turned in?
+                  // Thoughts, this could use hidden state?
```

Comment:
I dunno... sure? My first inclination is to leave them in, but I have no good reason for that... of course, we don't want to include them in the grade calculation, so use hidden for that?